src/core/attribute.d

```diff
@@ -0,0 +1,64 @@
+/**
+ * This module contains UDA's (User Defined Attributes) either used in
+ * the runtime or special UDA's recognized by compiler.
+ *
+ * Copyright: Copyright Jacob Carlborg 2015.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   Jacob Carlborg
+ * Source:    $(DRUNTIMESRC core/_attribute.d)
+ */
+
+/*          Copyright Jacob Carlborg 2015.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module core.attribute;
+
+/**
+ * This enum represent a meta attribute. Any declaration that has this attribute
+ * attached to itself is to be considered an attribute. That declaration
+ * should only be used as an attribute and never on its own.
+ */
+enum attribute;
```

Comment:
I know that this is not needed. I think that the [dmd pull][1] also does not depend on this. 

I was thinking that it would be nice if we could tag types (enums, structs, etc...) used as UDAs with an attribute to restrict their usage and specify if they should be inherited similar to the way [System.AttributeTargets][2] is used in .NET. That way a `core.attribute` module could be used in the future to provide some pretty interesting functionality. E.g. one could make a `@special_func` that has a trasitive property (functions tagged with it can only call other functions marked with the same attribute (similar to @nogc, or @safe/trusted/system)).

This idea of mine is not related to this pull request, so probably `enum attribute;` (along with its usages) should be removed for now.

[1]: https://github.com/D-Programming-Language/dmd/pull/4321/files
[2]: https://msdn.microsoft.com/en-us/library/tw5zxet9.aspx